src/compatibility.ts

```diff
@@ -74,3 +75,24 @@ export const requiredHubFlagWithDeprecations = requiredHubFlag({
   aliases: ['targetdevhubusername'],
   deprecateAliases: true,
 });
+
+/**
+ * @deprecated
+ */
+export const arrayWithDeprecation = (options: Record<string, unknown> = {}) =>
+  Flags.string({
+    // populate passed options
+    ...options,
+    // overlay those options we want to own
+    deprecated: { message: messages.getMessage('arrayStyleFlagsDeprecated') },
+    multiple: true,
+    parse: async (input: string) => {
+      if (input.includes(',')) {
```

Comment:
likely inconsequential perf nit: rather than scanning the string twice, you could say

```ts
const output = input.split(',').map((i) => i.trim())
```
and then use the array length as your warning conditional before returning the output?